### ISSUE TITLE
Rolling File: Add ability to ship every hour

### ISSUE THREAD

**Issue by slav** [serilog/serilog#692]
It would be very nice if rolling file would have ability to roll over every hour, in addition to every day. We're doing so much logging that we're able to keep on one machine only several hours worth of data while it's being shipped to the cluster where logs are stored.

**Comment by nblumhardt** [comment]
Hi @slav - thanks for the note. In general we've encouraged forking/diversification of the Rolling File sink, since the intersection of the (many) features that can be added to the classic rolling file tend to result in a complex beast.

E.g.: https://www.nuget.org/packages/Serilog.Sinks.RollingFileAlternate.MultiPlatform/, https://www.nuget.org/packages/Serilog.Sinks.RollingFileAlternate/, https://www.nuget.org/packages/Serilog.Sinks.RollingFile.Extension/.

Although it's likely "one day" the built-in rolling file will be extended (perhaps after 2.0 has shipped), in the short term grabbing a copy of the code and making a few changes to roll hourly might be the best bet. How would this work for you?

**Comment by slav** [comment]
@nblumhardt - In a way that's what I ended up doing. I created HourlyRollingFile that is hardcoded to roll every hour.

Then the next problem was that Durable sinks, such as ElasticSearchDurable use RollingFile, which is by day. So, I had to create a copy of ElasticSearchSink to use my HourlyRollingFile. So, the problem is getting compounded because RollingFile is used in other places.

Of course the problem now I won't be able to benefit from any updates to the ES sink or rolling file sink without manually doing maintenance of them.

That's why I thought since I've run into the issue and it's used in other places perhaps it would be better for RollingFile to be more flexibly. A lot of other sinks would benefit from it.
